client/src/utils/BGraphUtil.ts

```diff
@@ -125,6 +107,56 @@ export const executeBgraphEdges = (bgraphEdgeQuery: any, clause: Filter): any =>
   }
 };
 
+// Assume that bgraph instance passed in has already been initialized
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const executeBgraphPathQuery = (bgraphPathQuery: any, clause: Filter): any => {
+  switch (clause.field) {
+    // EDGE CLAUSES
+    case QUERY_FIELDS_MAP.BIO_PATH_PRE.field: {
+      return BGraphQueryUtil.bioPathPre(bgraphPathQuery, clause);
+    }
+    case QUERY_FIELDS_MAP.BIO_PATH_PRE_NODE_FILTER_LOOP.field: {
+      return BGraphQueryUtil.bioPathPreNodeFilterLoop(bgraphPathQuery, clause);
+    }
+    case QUERY_FIELDS_MAP.BIO_PATH_PRE_EDGE_FILTER_LOOP.field: {
+      return BGraphQueryUtil.bioPathPreEdgeFilterLoop(bgraphPathQuery, clause);
+    }
+    case QUERY_FIELDS_MAP.BIO_PATH_POST.field: {
```

Comment:
up to you but i would choose to put this case last to mirror the order in which these cases might be executed. thats just personal preference tho.